### PR TITLE
feat: simple coqui-ai stt integration

### PR DESCRIPTION
**Pull request by johannbarbie**
### What type of change does this PR introduce?
- [ ] Bugfix
- [x] Feature
- [ ] Refactor
- [ ] Documentation
- [ ] Not Sure?

### Does this PR introduce breaking changes?
- [x] Yes
- [ ] No

### List any relevant issue numbers:
- https://github.com/leon-ai/leon/issues/329
- https://github.com/leon-ai/leon/issues/339

### Description:

This adds Coqui for offline STT.  The parser is a copy of the deepspeech version. It works for me with node 16 and mac.

I have no prior experience with Leon or Coqui STT, so this might be very irrelevant or incomplete, let me know.

### CODE REVIEW

File: scripts/check.js [hunk 1]
```diff
@@ -173,11 +173,11 @@ export default () => new Promise(async (resolve, reject) => {
...
 
     log.info('Offline STT')
-    if (!fs.existsSync(deepSpeechPath)) {
+    if (!fs.existsSync(coquiPath)) {
       report.can_offline_stt.v = false
-      log.warning(`Cannot find ${deepSpeechPath}. You can setup the offline STT by running: "npm run setup:offline-stt"`)
+      log.warning(`Cannot find ${coquiPath}. You can setup the offline STT by running: "npm run setup:offline-stt"`)
     } else {
-      log.success(`Found DeepSpeech language model at ${deepSpeechPath}`)
+      log.success(`Found DeepSpeech language model at ${coquiPath}`)
```
Comment (JRMeyer) [inline scripts/check.js:180]: nit: `s/DeepSpeech language model/Coqui acoustic model/`

Coqui (like DeepSpeech) has both an acoustic model and a language model. The acoustic model is `model.tflite` and the language model ends in `*.scorer`. Here you're checking for `coquiPath`, which has been set to the acoustic model

File: scripts/setup-offline/setup-stt.js [hunk 2]
```diff
@@ -10,23 +10,23 @@ import os from '@/helpers/os'
 export default () => new Promise(async (resolve, reject) => {
   log.info('Setting up offline speech-to-text...')
 
-  const destDeepSpeechFolder = 'bin/deepspeech'
+  const destCoquiFolder = 'bin/coqui'
   const tmpDir = 'scripts/tmp'
-  const deepSpeechVersion = '0.9.3'
+  const coquiVersion = '1.0.0'
```
Comment (JRMeyer) [inline scripts/setup-offline/setup-stt.js:15]: `s/1.0.0/1.1.0/`

File: server/src/stt/coqui-stt/parser.js [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+import log from '@/helpers/log'
+
+log.title('Coqui-ai Parser')
+
+const parser = { }
+let STT = { }
+
+/* istanbul ignore next */
+try {
+  STT = require('stt-gpu') // eslint-disable-line global-require, import/no-unresolved
```
Comment (JRMeyer) [inline server/src/stt/coqui-stt/parser.js:13]: Coqui only releases TFLite models, so there's no GPU needed. `stt-gpu` is depreciated. You can read more detail [here](https://github.com/coqui-ai/STT/discussions/1922).

File: scripts/setup-offline/setup-stt.js (hunk 2 above)
Comment (johannbarbie) [reply scripts/setup-offline/setup-stt.js]: i couldn't find a `v1.1.0` here: https://github.com/coqui-ai/STT-models/tree/main/english/coqui, so i've renamed the to `coquiModelVersion` to distinguish it from the release version.

File: server/src/stt/stt.js [hunk 4]
```diff
@@ -10,7 +10,8 @@ class Stt {
     this.providers = [
       'deepspeech',
```
Comment (JRMeyer) [inline server/src/stt/stt.js:11]: `s/deepspeech/coqui/`

**Comment by louistiti** [comment]
Hi @johannbarbie 👋,

I just tried and it works out of the box. That is neat, thanks 👍

The docs and [new landing page](https://getleon.ai/) will be updated in the coming minutes according to these changes.

@JRMeyer @reuben, thanks for your support!

**Comment by JRMeyer** [comment]
New landing page looks slick 😊👍

Is there somewhere I should upload a high res logo for Coqui?

**Comment by louistiti** [comment]
> 

Thanks! Would it be possible to send a SVG logo of Coqui on Discord?
Here is my Discord ID:
Louistiti 🇨🇳#7928

**Comment by louistiti** [comment]
All nodes have been updates as mentioned in https://github.com/leon-ai/leon/issues/329#issuecomment-1025452104.

@JRMeyer I used [this SVG](https://coqui.ai/mask-icon.svg) and added the Coqui color to it. Please let me know if you still prefer to change it. Same as the docs, if you'd like to add/change something, do not hesitate 😉

**Comment by JRMeyer** [comment]
looks great :D 

FYI - there are two French Coqui STT models [[1](https://coqui.ai/french/commonvoice-fr/v0.6), [2](https://coqui.ai/french/jaco-assistant/v0.0.1)] and our community is training more :)

**Comment by louistiti** [comment]
[💥 [1.0.0-beta.6] Replace DeepSpeech with Coqui STT](https://trello.com/c/TbtaTiZw/414-%F0%9F%92%A5-100-beta6-replace-deepspeech-with-coqui-stt)

**Pull request merged**